### PR TITLE
feat(app): update server, naming, auth

### PR DESCRIPTION
**Pull request by JeanBarriere**
- use express instead of the node default http server (from postgraphile docs)
- rename asyncy images/databases to storyscript [REVERTED FOR DB NAME]
  - plugin naming -> storyscript
  - Dockerfile naming
- better auth check that returns a 401
- changes prettier to eslint and linted to standardjs rules

### CODE REVIEW

**Comment by JeanBarriere** [comment]
Also, I've updated the package.json versions - added the `yarn.lock` file which indicates this massive line added to the PR (≃+1900 lines)

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
 FROM          node:alpine
-LABEL         description="The Asyncy GraphQL stack, based on PostGraphile https://graphile.org/postgraphile/"
+LABEL         description="The Storyscript GraphQL stack, based on PostGraphile https://graphile.org/postgraphile/"
 EXPOSE        5000
 ENTRYPOINT    ["node", "server.js"]
 
-RUN           npm install -g yarn@1.9.4
+RUN           npm install -g yarn@latest
```
Comment (wilzbach) [inline Dockerfile]: Let's lock to a specific version here for reproducability. Also, otherwise you can't easily rebuild the docker image as the hash will still be the same.

File: app/StoryscriptPlugin.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
-  builder.hook("inflection", (inflection, build) => {
+module.exports = function StoryscriptPlugin (builder) {
+  builder.hook('inflection', (inflection, build) => {
     return {
       ...inflection,
 
       // Don't turn 'AppDns' into 'AppDn'
-      _singularizedTableName(table) {
-        const tableName = this._tableName(table);
+      _singularizedTableName (table) {
```
Comment (wilzbach) [inline app/StoryscriptPlugin.js]: Why this style change?

File: app/StoryscriptPlugin.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
+        const tableName = this._tableName(table)
         if (tableName.match(/dns$/i)) {
-          return tableName;
+          return tableName
```
Comment (wilzbach) [inline app/StoryscriptPlugin.js]: what was wrong here? Unrelated change :/

File: app/server.js [hunk 4]
```diff
@@ -1,126 +1,117 @@
...
       // COALESCE(...) a number of different settings.
-      "jwt.claims.owner_uuid": row.owner_uuid,
-      ...(row.permissions
-        ? { "jwt.claims.permissions": row.permissions.join(",") }
-        : null)
-    };
+      'jwt.claims.owner_uuid': auth.owner_uuid,
+      'jwt.claims.permissions': (auth.permissions || []).join(',')
+    }
+  } else if (typeof auth === typeof undefined) {
```
Comment (wilzbach) [inline app/server.js]:
```suggestion
  } else if (typeof auth === 'undefined') {
```

File: app/server.js [hunk 5]
```diff
@@ -1,126 +1,117 @@
...
+    // undefined means we have an 'Authorization' header but no match in DB
+    var error = new Error('InvalidOrExpiredToken')
+    error.statusCode = 401
+    error.name = 'InvalidOrExpiredToken'
+    throw error
   }
```
Comment (wilzbach) [inline app/server.js]: Uhm shouldn't we also throw if the token can't be found? e.g. `auth === false`

File: app/server.js [hunk 6]
```diff
@@ -1,126 +1,117 @@
...
-    );
-    if (!row) {
-      throw new Error("Invalid or expired token");
-    }
-    return {
-      ...basePermissions,
+    role: 'asyncy_visitor'
+  }
 
+  const auth = await getAuthorization(req)
```
Comment (wilzbach) [inline app/server.js]: Why throw the exception as it used to and re-throw them with a 401 status code? This will very likely result in cleaner code.

File: app/StoryscriptPlugin.js (hunk 3 above)
Comment (JeanBarriere) [reply app/StoryscriptPlugin.js]: just linting

File: Dockerfile (hunk 1 above)
Comment (JeanBarriere) [reply Dockerfile]: Ok

File: app/server.js (hunk 4 above)
Comment (JeanBarriere) [reply app/server.js]: not a fan of a type checking into string

File: Dockerfile (hunk 1 above)
Comment (judepereira) [reply Dockerfile]: Always and always lock versions. Never latest.

File: LICENSE [hunk 7]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2019 Asyncy
```
Comment (judepereira) [inline LICENSE]: This is not to be changed. This is to be included verbatim.

File: app/package.json [hunk 8]
```diff
@@ -1,12 +1,40 @@
...
+  "author": "Storyscript",
+  "license": "Apache-2.0",
+  "bugs": {
+    "url": "https://github.com/storyscript/platform-graphql/issues"
+  },
+  "homepage": "https://github.com/storyscript/platform-graphql#readme",
   "dependencies": {
-    "pg": "7.4.3",
-    "postgraphile": "4.0.0-rc.4"
+    "express": "^4.17.1",
```
Comment (judepereira) [inline app/package.json]: Does the `^` mean any version above the specified version?

File: app/server.js (hunk 5 above)
Comment (JeanBarriere) [reply app/server.js]: `getAuthorization` can returns whether a false boolean or a row object. If I get your comment well, 
you would prefer having a payload with more infos inside ?

File: run [hunk 9]
```diff
@@ -1,10 +1,10 @@
 # Install the PostGraphile watch fixtures (development only!)
 psql -f node_modules/graphile-build-pg/res/watch-fixtures.sql asyncy
 
-# Build the `asyncy/graphql` image
-docker build -t asyncy/graphql .
+# Build the `storyscript/graphql` image
+docker build -t storyscript/graphql .
```
Comment (judepereira) [inline run]: Please revert back to `asyncy/`. This image is used in the stack-kubernetes repo.

File: LICENSE (hunk 7 above)
Comment (wilzbach) [reply LICENSE]: Are you sure, because this just looks like it's from a license template and about the license of _this_ repo.

File: app/package.json (hunk 8 above)
Comment (JeanBarriere) [reply app/package.json]: It means it'll install the specific version or the latest patch/minor version

File: run (hunk 9 above)
Comment (JeanBarriere) [reply run]: Ok

File: app/StoryscriptPlugin.js (hunk 3 above)
Comment (wilzbach) [reply app/StoryscriptPlugin.js]: Hmm, in this case you auto-linter destroys the entire git history :/

File: LICENSE (hunk 7 above)
Comment (JeanBarriere) [reply LICENSE]: Ok

File: app/package.json (hunk 8 above)
Comment (wilzbach) [reply app/package.json]: No everything until the next major version:

image

File: LICENSE (hunk 7 above)
Comment (judepereira) [reply LICENSE]: Looks like you're right.

File: app/package.json (hunk 8 above)
Comment (judepereira) [reply app/package.json]: Ah no no. In that case, let's not do that. Lock it to one minor and patch down too. While we understand the differences between minor and patch in the semver world, somebody else's interpretation might be different, and can cause unwanted headache for us in the future.

Comment (JeanBarriere) [reply app/package.json]: Yeah the latest minor/patch revision is used. 
In node, everyone is turning to use the semver `major.minor.patch` versioning today @judepereira, no headaches with packages like express, postgres or similars.

**Comment by JeanBarriere** [comment]
Okay, as there is a lot of controversial things here apparently, I'm moving this to a more concise PR

File: app/package.json (hunk 8 above)
Comment (judepereira) [reply app/package.json]: I've had bad experience with people even trying to follow semver. Lets please lock down versions completely.

**Comment by JeanBarriere** [comment]
Moving to #7
